lib/Onnxifi/Base.h

```diff
@@ -112,9 +154,13 @@ class Graph {
   /// Run inference.
   onnxStatus run();
 
+  /// Run async using thread pool.
+  static void runAsync(std::function<void(void)> fn);
+
 private:
   BackendPtr backendPtr_;
   Function *function_;
+  static ThreadPool threadPool_;
```

Comment:
Yeah, I think this is pretty reasonable.  But to avoid SIOF, I'd be inclined to do a Meyers singleton like, 
```
ThreadPool &getThreadPool() {
  static ThreadPool threadPool(kNumWorkers);
  return threadPool;
}
```